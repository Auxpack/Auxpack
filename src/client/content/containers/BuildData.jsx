import React from 'react'

const BuildData = ({ build }) => {

    console.log(`build: `, build)
    return (
        <h2>Build Data</h2>
    );
}

export default BuildData;