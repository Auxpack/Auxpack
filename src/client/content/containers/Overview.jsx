import React from 'react'
import SunburstContainer from './SunburstContainer.jsx'

const Overview = ({ build, activeBuild }) => {
    //overview container for information if we wanted more than the sunburst on the first page
    return (
        <div id="container">
            <SunburstContainer build={build} activeBuild={activeBuild} />
        </div>
    );
}

export default Overview;