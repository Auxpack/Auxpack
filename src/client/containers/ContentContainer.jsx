import React from 'react';
import { Switch, Route } from 'react-router-dom';
import Overview from '../content/containers/Overview.jsx';
import BuildData from '../content/containers/BuildData.jsx';
import TreeShaking from '../content/containers/TreeShaking.jsx';
import HistoryCharts from '../content/containers/HistoryCharts.jsx';

const ContentContainer = (props) => {
    const { build, activeBuild, handleInc, handleDec } = props;
    return (
        <React.Fragment>
            <Switch>
                <Route
                    exact path="/"
                    render={() => (
                        <Overview
                         build={build} 
                         activeBuild={activeBuild} 
                        />
                    )}
                />
                <Route
                    exact path="/build"
                    render={() => (
                        <BuildData
                            build={build}
                            activeBuild={activeBuild}
                            handleDec={handleDec}
                            handleInc={handleInc}
                        />
                    )}
                />
                <Route
                    exact path="/treeshaking"
                    render={() => (
                        <TreeShaking
                        build={build}
                        activeBuild={activeBuild}
                        />
                    )}
                />
                <Route
                    exact path="/history"
                    render={() => (
                        <HistoryCharts
                        build={build}
                        />
                    )}
                />
            </Switch>
        </React.Fragment>
    );
}

export default ContentContainer;