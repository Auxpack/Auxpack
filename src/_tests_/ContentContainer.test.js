import ContentContainer from '../client/containers/ContentContainer.jsx';
import Overview from '../client/content/containers/Overview.jsx';
import BuildData from '../client/content/containers/BuildData.jsx'
import Treeshaking from '../client/content/containers/Treeshaking.jsx'

describe('ContentContainer Unit Tests', () => {
    let wrapper;

    const props = {
        build: [{
            timeStamp: 1575426090404,
            time: 1439,
            hash: "546142ce1b49a6ba7d6f",
            errors: [],
            size: 1172113,
            assets: [{ "name": "bundle.js", "chunks": ["main"], "size": 1172113 }],
            chunks: [{ "size": 1118609, "files": ["bundle.js"], "modules": [{ "name": "./client/App.jsx", "size": 6375, "id": "./client/App.jsx" }] }],
            treeStats: { csj: [], esm: [], both: [] }
        }],
    }

    beforeEach(() => {
        wrapper = shallow(<ContentContainer {...props} />);
    })

    it('should render', () => {
        expect(wrapper);
    })

    test('ContentContainer snapshot testing', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    xit('ContentContainer should render Overview', () => {
        //console.log(mount(<ContentContainer />).debug());
        expect(wrapper.find(Overview).length).to.equal(1);
    })
})