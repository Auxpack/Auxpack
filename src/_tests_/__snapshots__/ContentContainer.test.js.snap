// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentContainer Unit Tests ContentContainer snapshot testing 1`] = `
<Fragment>
  <Router>
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/build"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/treeshaking"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/history"
        render={[Function]}
      />
    </Switch>
  </Router>
</Fragment>
`;
